frontend/src/pages/modelServing/screens/metrics/ScheduledMetricSelect.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Select, SelectOption } from '@patternfly/react-core';
+
+type ScheduledMetricSelectProps = {
+  selected: string | undefined;
+  options: string[];
+  onSelect: (name: string) => void;
+};
+const ScheduledMetricSelect: React.FC<ScheduledMetricSelectProps> = ({
+  selected,
+  options,
+  onSelect,
+}) => {
+  const [isOpen, setOpen] = React.useState(false);
+
+  const onToggle = () => setOpen(!isOpen);
+  const onSelectValue = (event, selection) => {
+    onSelect(selection);
+    setOpen(false);
+  };
```

Comment:
sounds good, happy to change. Out of curiosity if 'event' is the first param and I just omit it, will TS do some stuff to under the hood it make it work in the compiled JS?

Reply:
Not quite -- you have two choices imo (three if you disregard my preference).

First is inline for ease of use and (imo) readability:
```
<Component onSomething=((event, selection) => { ... }} />
```
`event, selection` will type based on inference from the component's definition of `onSomething`. 

Second is you can simplify and coordinate:
```
const handleSomething = (selection: string) => { ... }

<Component onSomething={(event, selection) => handleSomething(selection)} />
```
You'll likely have to manage some type stuff with selection, but it should be narrowed before you kick it out. 

Third is what you're doing. Where you separate the two and need to type it fully. I personally don't recommend this, because you end up writing types that you could just infer. 

I have a stronger preference to making things very straight forward and inline. If you want to separate it, make it easier so your separation is not a mirror of what is in PF. TS can infer and "pull" types if you need to. But pulling params is iffy, and not usually worth it.